fs-cache/src/memory_limited_storage.rs

```diff
@@ -0,0 +1,273 @@
+                "Path is not a directory".to_owned(),
+            ));
+        }
+
+        // First pass: collect metadata only
+        let mut file_metadata = Vec::new();
+        for entry in fs::read_dir(&self.path)? {
+            let entry = entry?;
+            let path = entry.path();
+            if path.is_file()
+                && path
+                    .extension()
+                    .map_or(false, |ext| ext == "json")
+            {
+                if let Ok(metadata) = fs::metadata(&path) {
+                    let key = extract_key_from_file_path(&self.label, &path)?;
+                    file_metadata.push((key, metadata.len() as usize));
+                }
+            }
+        }
+
+        // Sort by timestamp (newest first) before loading any values
+        file_metadata.sort_by(|a, b| b.1.cmp(&a.1));
+
+        // Clear existing cache
+        self.memory_cache.clear();
+        self.current_memory_bytes = 0;
+
+        // TODO: Need some work here
+        // Second pass: load only the values that will fit in memory
+        let mut loaded_bytes = 0;
+        let mut total_bytes = 0;
+
+        for (key, approx_size) in file_metadata {
+            total_bytes += approx_size;
+
+            // Only load value if it will likely fit in memory
+            if loaded_bytes + approx_size <= self.max_memory_bytes {
+                match self.load_value_from_disk(&key) {
+                    Ok(value) => {
+                        let actual_size = Self::estimate_size(&value);
+                        if loaded_bytes + actual_size <= self.max_memory_bytes {
+                            self.memory_cache.insert(key, value);
+                            loaded_bytes += actual_size;
+                        }
+                    }
+                    Err(err) => {
+                        log::warn!(
+                            "{} cache: failed to load key={}: {}",
+                            self.label,
+                            key,
+                            err
+                        );
+                    }
+                }
+            }
+        }
+
+        self.current_memory_bytes = loaded_bytes;
+
+        log::debug!(
+            "{} loaded {}/{} bytes in memory",
+            self.label,
+            self.current_memory_bytes,
+            total_bytes
+        );
+
+        Ok(())
+    }
+
+    fn estimate_size(value: &V) -> usize {
+        serde_json::to_vec(value)
+            .map(|v| v.len())
+            .unwrap_or(0)
+    }
+
+    // Write a single value to disk
+    fn write_value_to_disk(&mut self, key: &K, value: &V) -> Result<()> {
+        let file_path = self.path.join(format!("{}.json", key));
+        let mut file = File::create(&file_path)?;
+        file.write_all(serde_json::to_string_pretty(&value)?.as_bytes())?;
+        file.flush()?;
+
+        let new_timestamp = SystemTime::now();
+        file.set_modified(new_timestamp)?;
+        file.sync_all()?;
```

Comment:
Do we really need it?

Reply:
No, not needed anymore, since we are no longer doing any kind of synchronization.

Reply:
I think we may need it, in `fn load_fs`, we compare timestamps to load recent files into memory. This could help us efficiently decide which files to load or skip since it provides precise timing. However, it's a very minor improvement, so we can skip it.